src/lib.rs

```diff
@@ -169,48 +170,39 @@ pub fn filter_python_executables(
             continue;
         }
         let version_part = &file_name["python".len()..];
-        if let Ok(found_version) = RequestedVersion::from_string(&version_part.to_string()) {
+        if let Ok(found_version) = RequestedVersion::from_str(version_part) {
             match found_version {
-                RequestedVersion::Exact(major, minor) => executables.insert(
-                    Version {
-                        major: major,
-                        minor: minor,
-                    },
-                    path.clone(),
-                ),
+                RequestedVersion::Exact(major, minor) => {
+                    executables.insert(Version { major, minor }, path.clone())
+                }
                 _ => continue,
             };
         }
     }
 
-    return executables;
+    executables
```

Comment:
Is it common to leave out `return` even in the case of using a dangling reference for the return value? Just looks odd to me, hence the question. 😄

Reply:
I'm not sure I understand you exactly. When we return the `HashMap`, it is moved, not copied and not passed by reference. With or without `return`, it's precisely the same. I believe it's more idiomatic to ommit it when possible (although I'm not entirelly convinced it's better).